document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -270,18 +276,32 @@ protected String parseQueryForCompositeFilter(Filter filter) {
       case AND: {
         String childList =
             Arrays.stream(filter.getChildFilters())
-                .map(this::parseQuery)
+                .map(childFilter -> parseFilter(childFilter, paramsBuilder))
                 .filter(str -> !StringUtils.isEmpty(str))
                 .map(str -> "(" + str + ")")
                 .collect(Collectors.joining(" AND "));
         return !childList.isEmpty() ? childList : null;
       }
       default:
         throw new UnsupportedOperationException(
-            String.format("Boolean operation:%s not supported", op));
+            String.format("Query operation:%s not supported", op));
     }
   }
 
+  @VisibleForTesting
+  protected PreparedStatement buildPreparedStatement(String sqlQuery, Params params) throws SQLException {
+    PreparedStatement preparedStatement = client.prepareStatement(sqlQuery);
+    params.getStringParams().forEach((k, v) -> {
```

Comment:
Based on the suggestion above on using `Map<Integer, Object>` for param, we can do as below.
```
protected PreparedStatement buildPreparedStatement(String sqlQuery, Params params) throws SQLException {
    PreparedStatement preparedStatement = client.prepareStatement(sqlQuery);
    params.getObjectParams().forEach((k, v) -> {
      try {
        // Postgres Index starts from 1
        isValidTypes(value) // true if type is string, int, long, double, false for composite type
        preparedStatement.setString(k, String.valueOf(value));
      } catch (SQLException e) {
        LOGGER.error("SQLException querying documents. query: {}", sqlQuery, e);
      }
    });
    return preparedStatement;
  }
```
A drawback of this approach is that for `>` operator, we might have a `string` value. And, some document-store might have support for this as well. So, we can just check valid types for now - string, int, long, double and discard composite Object types. What do you guys think?

Reply:
If we want to check basic types, why even change it to a generic Map<Integer, Object>?

Reply:
if we go with the existing definition, we will have to add casting logic while filter parsing. The `INT` value of the filter, goes into Map<Interger, Integer>. But, I don't see how we can use this specific information while doing a query. We still have to convert all of them into finally `String` as we don't know the type of destination field in `jsonb` document. So, suggesting that we can just simplify logic by replacing all Where clause filter with `?` and collecting all values into Map<Integer, Object>.